### PR TITLE
Bug 1217259 - Remove audio-competing hacks from the Ringtones app

### PR DESCRIPTION
**Pull request by jimporter**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e334dd3de5eb684bf47cea1c516806e23bbef06b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e334dd3de5eb684bf47cea1c516806e23bbef06b

File: apps/ringtones/js/tone_player.js [hunk 1]
```diff
@@ -131,28 +127,4 @@ TonePlayer.prototype = {
...
-   * from playing once we've started previewing ringtones, or destroys the
-   * context.
-   *
-   * @param {Boolean} exclusive true to create the AudioContext, false to
-   *   destroy.
-   */
-  _setExclusiveMode: function(exclusive) {
-    if (exclusive) {
-      if (!this._source) {
-        this._context = new AudioContext('ringer');
```
Comment (davidflanagan) [inline apps/ringtones/js/tone_player.js]: Should you also remove the 'audio-channel-ringer' permission from the manifest? 

And don't you need to set mozAudioChannelType = 'content' somewhere now that you're taking this out?  Without that won't the ringtone just play over the top of background music? And I suppose then you need audio-channel-content permission?

File: apps/ringtones/js/share.js [hunk 2]
```diff
@@ -126,60 +126,8 @@ function handleShare(data, callback) {
...
-          }
-          else {
-            preview.play();
-          }
-        }());
-        // END OF HACK
-        // When there is a real bug fix, just replace this entire if clause
-        // with:
-        //   preview.play();
-      }
```
Comment (davidflanagan) [inline apps/ringtones/js/share.js]: That was a cool hack :-)

I'm surprised that you're not removing the music app side of it in this same patch.  (Or is the music side already removed in the NGA code?)

Comment (jimporter) [inline apps/ringtones/js/share.js]: We never added it in NGA, so it turns out the code was working this whole time without the hack anyway. :)

File: apps/ringtones/js/tone_player.js (hunk 1 above)
Comment (jimporter) [inline apps/ringtones/js/tone_player.js]: I'll take a look at this, but the behavior seems to work as expected right now.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a3d85ca551cc9260bb88d1801c72364413ef1111

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7dbdc340136d4f74d57a1c61906c38868285a396

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a62eff48c6942707188c869fb118994194a21c40

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d168c75279b5fc57286f4e426b2b82c77d3ee070

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=35e28ad75659b0e473fc3f1d96d4894b3215226a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=53b6700c91ce66d0ef8be36939649fb900424f4d

**Pull request closed without merging**